### ISSUE TITLE
`prefer-active-doc` false positive on `constructor` (prototype pollution bug)

### ISSUE THREAD

**Issue by gapmiss** [obsidianmd/eslint-plugin#128]
Discovered and written by CLAUDE, while we were updating/testing https://github.com/gapmiss/obsidian-plugin-skill:

---

The rule uses direct property access on the `REPLACEMENTS` object without checking `hasOwnProperty`, causing false positives when identifiers match `Object.prototype` property names.

**Root cause:** Line 29 in `preferActiveDoc.js`:
```js
const replacement = REPLACEMENTS[node.name];
```

When `node.name === "constructor"`, this returns `Object.prototype.constructor` instead of `undefined`.

**Fix:**
```js
const replacement = Object.hasOwn(REPLACEMENTS, node.name) ? REPLACEMENTS[node.name] : undefined;
```

**Affected identifiers:** `constructor`, `toString`, `valueOf`, `hasOwnProperty`, `__proto__`, etc.

---

**Comment by saberzero1** [comment]
https://github.com/obsidianmd/eslint-plugin/pull/126

**Comment by mnaoumov** [comment]
Addressed in #126 + #129
